substrate/frame/system/src/extensions/check_weight.rs

```diff
@@ -136,8 +130,27 @@ where
 	}
 }
 
+/// Check that the combined extrinsic length and proof size together do not exceed the PoV limit.
+pub fn check_combined_proof_size(
+	maximum_weight: &BlockWeights,
+	next_len: u32,
+	next_weight: &crate::ConsumedWeight,
+) -> Result<(), TransactionValidityError> {
+	// This extra check ensures that the extrinsic length does not push the
+	// PoV over the limit.
+	let total_pov_size = next_weight.total().proof_size().saturating_add(next_len as u64);
+	if total_pov_size > maximum_weight.max_block.proof_size() {
```

Comment:
It is just checking against `max_block` here instead of `per_class` for the class of the extrinsic?

Reply:
Yes, that is intentional. We are still checking the proof and ref_time weight against the extrinsic class limits. In addition the block_length total, which has no classes. This additional check should just make sure that the total limit is respected.